data/data_loader.py

```diff
@@ -41,40 +42,27 @@ def __init__(self, image_train_items: list[ImageTrainItem], seed=555, batch_size
         self.rating_overall_sum: float = 0.0
         self.ratings_summed: list[float] = []
         self.__update_rating_sums()
+        count_including_multipliers = sum([math.floor(max(i.multiplier, 1)) for i in self.prepared_train_data])
+        if count_including_multipliers > len(self.prepared_train_data):
+            logging.info(f" * DLMA initialized with {len(image_train_items)} items ({count_including_multipliers} items total after applying multipliers)")
+        else:
+            logging.info(f" * DLMA initialized with {len(image_train_items)} items")
+
 
     def __pick_multiplied_set(self, randomizer):
         """
         Deals with multiply.txt whole and fractional numbers
         """
-        #print(f"Picking multiplied set from {len(self.prepared_train_data)}")
-        data_copy = copy.deepcopy(self.prepared_train_data) # deep copy to avoid modifying original multiplier property
-        epoch_size = len(self.prepared_train_data)
         picked_images = []
-
-        # add by whole number part first and decrement multiplier in copy
-        for iti in data_copy:
-            #print(f"check for whole number {iti.multiplier}: {iti.pathname}, remaining {iti.multiplier}")
-            while iti.multiplier >= 1.0:
+        for iti in self.prepared_train_data:
+            multiplier = iti.multiplier
+            while multiplier >= 1:
                 picked_images.append(iti)
-                #print(f"Adding {iti.multiplier}: {iti.pathname}, remaining {iti.multiplier}, , datalen: {len(picked_images)}")
-                iti.multiplier -= 1.0
-
-        remaining = epoch_size - len(picked_images)
-
-        assert remaining >= 0, "Something went wrong with the multiplier calculation"
-
-        # add by remaining fractional numbers by random chance
-        while remaining > 0:
-            for iti in data_copy:
-                if randomizer.uniform(0.0, 1.0) < iti.multiplier:
-                    #print(f"Adding {iti.multiplier}: {iti.pathname}, remaining {remaining}, datalen: {len(data_copy)}")
-                    picked_images.append(iti)
-                    remaining -= 1
-                    iti.multiplier = 0.0
-                if remaining <= 0:
-                    break
-        
-        del data_copy
+                multiplier -= 1
+            # deal with fractional remainder
+            if multiplier > randomizer.uniform(0, 1):
```

Comment:
Hmm. Ok actually there's still a problem with the logic here. I don't have more time to fix this now, will do it later. Basically this needs to a) ensure the epoch size stays the same and b) pick a subset per-folder based on the folder's shared multiplier.